### ISSUE TITLE
Add a note in Scaladoc about queue capacity

### ISSUE THREAD

**Issue by fblundun** [snowplow/snowplow-scala-tracker#16]
http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/LinkedBlockingQueue.html

At the moment the LinkedBlockingQueue we use has maximum size `Int.MaxValue`. If an application is creating events faster than the queue handles them we could conceivably get an OutOfMemoryError as too many items are simultaneously waiting in the queue.

We could get around this by reducing the maximum size of the queue or letting users pass in their own queue.

I don't think this is very urgent - an application would have to be creating a lot of events to cause any problems.

**Comment by chuwy** [comment]
At least we can warn users and propose to use batch emitter because it's queue contains `Seq[Map[String, String]]`.

**Comment by fblundun** [comment]
Adding a note about this to the wiki sounds good.

**Comment by chuwy** [comment]
Pushing back to 0.4.0, it depends on #22.

**Comment by chuwy** [comment]
I think it wontfix. 

1. Queue size is really big enough
2. It's especially big for emitter with buffer
3. It's easy enough to add custom emitter backed by some super-capacious queue (like celery queue in python tracker)
4. As far as I see, it won't exit with OOM, but just block thread

I will just add a note in ScalaDoc.
